docs/cicd-with-bicep.md

```diff
@@ -0,0 +1,68 @@
+# Adding bicep to a CI/CD pipeline
+
+As your bicep practice matures, you will want to check-in your bicep code into source control and kick off a pipeline or workflow, which would do the following:
+
+1. Build your bicep file into an ARM Template
+1. Deploy the generated ARM template
+
+In order to do this, we need to make sure the bicep CLI is installed on the build agent. For now, bicep is not preinstalled on any build agents or tasks provided by Microsoft, but installing it manually as part of the pipeline is straightforward.
+
+The following example is designed to be run in GitHub actions workflow and uses Azure CLI, but could be easily adapted to run in a Azure DevOps Pipeline. It assumes the following prerequisite:
+
+* The bicep file you want to transpile and deploy is called `main.bicep` and exists in the root of the repo
+* You are deploying the transpiled ARM Template to a resource group. Deploying to another scope like a subscription requires a different CLI command.
+
+```yaml
+
+name: bicep build and deploy
+
+on: push
+
+jobs:
+  bicep-build-and-deploy:
+    name: bicep build and deploy
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checks out a copy of your repository on the ubuntu-latest machine
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      # Install the latest release of the bicep CLI
+      - name: Install bicep CLI
+        run: |
+          curl -Lo bicep https://github.com/Azure/bicep/releases/latest/download/bicep-linux-x64
+          chmod +x ./bicep
+          sudo mv ./bicep /usr/local/bin/bicep
+          bicep --help
```

Comment:
Nice! Do you think it's also worth showing an example of CI (vs CD) - just the build step?

Reply:
meaning have a separate example that just goes to line 42?

Reply:
Exactly. Feels trivially similar, but also serves a different purpose. I'm in two minds about whether it's worthwhile to include.

Reply:
I added a comment to point out where "CI" transitions into "CD", but for now I feel like it's overkill to go with a separate file